rosidl_typesupport_c/QUALITY_DECLARATION.md

```diff
@@ -0,0 +1,99 @@
+
+### Public API Declaration
+
+All symbols in the installed headers are considered part of the public API.
+
+All installed headers are in the `include` directory of the package, headers in any other folders are not installed and considered private.
+
+## Change Control Process
+
+`rosidl_typesupport_c` follows the recommended guidelines for ROS Core packages in the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#package-requirements).
+
+This includes:
+
+- all changes occur through a pull request
+- all pull request have at least 1 peer review
+- all pull request must pass CI on all [tier 1 platforms](https://www.ros.org/reps/rep-2000.html#support-tiers)
+- all pull request must resolve related documentation changes before merging
+
+## Documentation
+
+### Feature Documentation
+
+`rosidl_typesupport_c` does not have any feature documentation and it will need to be added for higher quality levels.
+
+### Public API Documentation
+
+`rosidl_typesupport_c` does not currently have public API documentation.
+
+### License
+
+The license for `rosidl_typesupport_c` is Apache 2.0, and a summary is in each source file, the type is declared in the `package.xml` manifest file, and a full copy of the license is in the `LICENSE` file.
+
+There is an automated test which runs a linter that ensures each file has a license statement.
+
+### Copyright Statements
+
+The copyright holders each provide a statement of copyright in each source code file in `rosidl_typesupport_c`.
+
+There is an automated test which runs a linter that ensures each file has at least one copyright statement.
+
+## Testing
+
+### Feature Testing
+
+There are currently no public features undergoing tests.
+
+### Public API Testing
+
+There are currently no tests for the public API.
+
+### Coverage
+
+`rosidl_typesupport_c` does not currently track test coverage.
+
+### Performance
+
+`rosidl_typesupport_c` does not currently have performance tests.
+
+### Linters and Static Analysis
+
+`rosidl_typesupport_c` uses and passes all the standard linters and static analysis tools for a C++ package as described in the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#linters).
+
+## Dependencies
+
+`rosidl_typesupport_c` has the following ROS dependencies:
+* `rcpputils`
+* `rosidl_runtime_c`
+* `rosidl_typesupport_connext_c`
+* `rosidl_typesupport_interface`
+* `rosidl_typesupport_introspection_c`
+
+It has "buildtool" dependencies, which do not affect the resulting quality of the package, because they do not contribute to the public library API.
+It also has several test dependencies, which do not affect the resulting quality of the package, because they are only used to build and run the test code.
+
+## Platform Support
+
+`rosidl_typesupport_c` supports all of the tier 1 platforms as described in [REP-2000](https://www.ros.org/reps/rep-2000.html#support-tiers), and tests each change against all of them.
```

Comment:
A link to test results would also be nice here.

Reply:
Added release nightlies here